flask_parameter_validation/docs_blueprint.py

```diff
@@ -133,11 +155,240 @@ def docs_json():
+                subschema["type"] = "string"
+                subschema["format"] = "date-time"
+                if "datetime_format" in param["loc_args"]:
+                    warnings.warn("datetime_format cannot be translated to JSON Schema, please use ISO8601 date-time",
+                                  Warning, stacklevel=2)
+            elif p in ["date", "datetime.date"]:
+                subschema["type"] = "string"
+                subschema["format"] = "date"
+            elif p in ["time", "datetime.time"]:
+                subschema["type"] = "string"
+                subschema["format"] = "time"
+            elif p == "dict":
+                subschema["type"] = "object"
+            elif p in ["None", "NoneType"]:
+                subschema["type"] = "null"
+            elif p in ["StrEnum", "IntEnum"]:
+                if p == "StrEnum":
+                    subschema["type"] = "string"
+                elif p == "IntEnum":
+                    subschema["type"] = "integer"
+                subschema["enum"] = param["enum_values"]
+            else:
+                warnings.warn(f"generate_json_schema_helper received an unexpected parameter type: {p}",
+                              Warning, stacklevel=2)
+            schemas.append(subschema)
+        if len(schemas) == 1 and parent_group is None:
+            return schemas[0]
+        elif parent_group in ["Optional", "Union"]:
+            return {"oneOf": schemas}
+        elif parent_group in ["List", "list"]:
+            schema = {"type": "array", "items": schemas[0]}
+            if "min_list_length" in param["loc_args"]:
+                schema["minItems"] = param["loc_args"]["min_list_length"]
+            if "max_list_length" in param["loc_args"]:
+                schema["maxItems"] = param["loc_args"]["max_list_length"]
+            return schema
+        else:
+            warnings.warn(f"generate_json_schema_helper encountered an unexpected type: {param_type} with parent: "
+                          f"{parent_group}", Warning, stacklevel=2)
+
+
+def generate_json_schema_for_parameter(param):
+    """ Generate JSON Schema for a single parameter """
+    return generate_json_schema_helper(param, param["type"])
+
+
+def generate_json_schema_for_parameters(params):
+    """ Generate JSON Schema for all parameters of a route"""
+    schema = {
+        "type": "object",
+        "properties": {},
+        "required": []
+    }
+    for p in params:
+        schema_parameter_name = p["name"] if "alias" not in p["loc_args"] else p["loc_args"]["alias"]
+        if "json_schema" in p["loc_args"]:
+            schema["properties"][schema_parameter_name] = p["loc_args"]["json_schema"]
+        else:
+            schema["properties"][schema_parameter_name] = generate_json_schema_for_parameter(p)
+        if parameter_required(p):
+            schema["required"].append(schema_parameter_name)
+    return schema
+
+
+def generate_openapi_paths_object():
```

Comment:
Please can we add some comments/spacing to this function? It is quite tricky to understand the flow atm. Otherwise looks great!